File: domain/lib/meeting/meeting.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'meeting.freezed.dart';

@freezed
class Meeting with _$Meeting {
  factory Meeting({
    required Map<String, bool> participantIds,
    required Map<String, String> participantNames,
    required Map<String, String> participantUrls,
    required String id,
    required String name,
    required DateTime time,
    required String createdUserId,
    required int participantCount,
  }) = _Meeting;
}

File: functions/java/src/main/java/functions/Meeting.kt
package functions

import java.util.*

data class Meeting(
    val participantIds: Map<String, Boolean> = mapOf(),
    val participantNames: Map<String, String> = mapOf(),
    val participantUrls: Map<String, String> = mapOf(),
    val id: String = "",
    val name: String = "",
    val time: Date = Date(),
    val createdUserId: String = "",
    val participantCount: Long = 0,
)